R/air.R

```diff
@@ -0,0 +1,213 @@
+#' Configure a project to use Air
+#'
+#' @description
+#' [Air](https://github.com/posit-dev/air) is an extremely fast R code
+#' formatter. This function sets up a project to use Air. Specifically, it:
+#'
+#' - Creates an empty `air.toml` configuration file. If either an `air.toml` or
+#'   `.air.toml` file already existed, nothing is changed. If the project is an
+#'   R package, `.Rbuildignore` is updated to ignore this file.
+#'
+#' - Adds recommended settings to `.vscode/settings.json` and
+#'   `.vscode/extensions.json`. This includes:
+#'
+#'   - Setting `editor.formatOnSave = true` for R files to enable formatting on
+#'     every save. Does nothing if `editor.formatOnSave` was already set.
```

Comment:
There is something of an existing convention where we use `can_overwrite()` to determine if it's safe to err on the side of making change in a file. If the user has set the `usethis.overwrite` option to `TRUE` and we're making changes inside a project that is a Git repo, we just go ahead and do it. I have this set to `TRUE` in my `.Rprofile`.

I'm not sure if you'd like to do that here, but just wanted you to know about this pattern.

Reply:
I think I'd use that if we were completely overwriting. In this particular case, we want to _merge_ with what the user already has as much as possible, so I think not asking or checking is ok in this case.

I do think I'm going to remove this though

```
Does nothing if `editor.formatOnSave` was already set.
Does nothing if `editor.defaultFormatter` was already set.
```

I think we actually should update this, as you've explicitly requested Air with `use_air()`, so you've opted into these recommended settings already

Reply:
Yeah when I first looked at this I thought maybe it was overly cautious / overly respectful of existing settings. Happy to have it just make these changes unconditionally.